utils/billgeneration.js

```diff
@@ -146,19 +173,49 @@ end date: ${endDate}`
     status: 'Pending'
   });
 
+  // Get computed Fields
+  const newBill = await getSubscriberBillById(newBillId);
+
   // Generate PDF!
-  generatePdf(subscriber, solarProject, newBillId, prevBill.id);
+  let localPdfPath;
+  try {
+    localPdfPath = await generatePdf(
+      subscriber,
+      solarProject,
+      newBill,
+      prevBill.id
+    );
+  } catch (e) {
+    console.log(e);
+    console.log('Run into error in PDF Generation process. Reporting...');
+    sendEmail(pdfBillError(subscriber, solarProject, e.message));
+    return;
+  }
+
+  // Report Success
+  const approveLink = `${Constants.SERVER_URL}/approve?id=${newBill.id}`;
+  sendEmail(
+    billSuccess(subscriber, solarProject, newBill, approveLink, localPdfPath)
+  );
```

Comment:
is there a reason we're using this instead of generatePdf as before?

Reply:
generatePdf is still in there! Line 182